src/wp-includes/l10n.php

```diff
@@ -866,8 +925,9 @@ function unload_textdomain( $domain, $reloadable = false ) {
 
 		unset( $l10n[ $domain ] );
 
+		// Since we support multiple locales, we don't actually need to unload reloadable text domains.
 		if ( ! $reloadable ) {
-			$l10n_unloaded[ $domain ] = true;
+			return WP_I18n_Translation_Controller::instance()->unload( $domain );
 		}
```

Comment:
I am seeing throughout this PR you no longer set `$l10n_unloaded[ $domain ] = true;`, also via `unload_textdomain( ..., true )`. Does that effectively mean we never unload any text domains? What defines whether a text domain is reloadable or not?

Reply:
`$l10n_unloaded[ $domain ] = true;` is still called in some places.

> What defines whether a text domain is reloadable or not?

This is up to the caller of `unload_textdomain()`, `$reloadable` is the second 

The whole `$reloadable` concept is related to the locale switching feature in WordPress. The problem there was that WordPress could only load 1 translation at a time. So when switching locales, translations had to be unloaded but you'd still need to be able to load them again just-in-time. There's also test coverage around this logic.

Reply:
@swissspidy I'm not sure this answers my question. I understand the function signature, but what's a use-case where a locale should _not_ be reloadable?

Reply:
I don‘t know off the top of my head if we use that in core, but some plugins might want to force unload without it being able to be loaded again accidentally. I‘d need to dig up the old ticket to find examples.

Reply:
btw I just restored that line in question here for better back compat with that global.

Reply:
> btw I just restored that line in question here for better back compat with that global.

Is that correct though? For example, in `unload_textdomain()`, the `$l10n_unloaded[ $domain ] = true;` is only applied if the translation is _not_ reloadable. But as far as I can tell, per the logic here it's always reloadable, so I think we would instead need to `unset( $l10n_unloaded[ $domain ] );`, if anything.

Reply:
> I don‘t know off the top of my head if we use that in core, but some plugins might want to force unload without it being able to be loaded again accidentally. I‘d need to dig up the old ticket to find examples.

That would be good. I mostly want to know _why_. _Why_ would someone want to prevent a translation file from being reloaded?

Reply:
OK just looked at it again. So, `unload_textdomain()` used to unload textdomains completely, preventing them from being reloaded again. That was and is the default behavior. Then we introduced locale switching.

And when we switch locales, we used to need to unload existing translations, load new ones, and then unload those again and load the original ones again. Hence we added the new `$reloadable` param to support this in a back compat way.

Now while we still need to support the original behavior, `$reloadable` becomes easier because we don't actually unload in that case, we can just store multiple languages in memory.

Makes sense?